api/utils/verify.js

```diff
@@ -0,0 +1,57 @@
+'use strict';
+var crypto = require('crypto');
+
+//Credits to https://ciphertrick.com/2016/01/18/salt-hash-passwords-using-nodejs-crypto/
+
+/**
+ * generates random string of characters i.e salt
+ * @function
+ * @param {number} length - Length of the random string.
+ */
+var genRandomString = function(length){
+    return crypto.randomBytes(Math.ceil(length/2))
+            .toString('hex') /** convert to hexadecimal format */
+            .slice(0,length);   /** return required number of characters */
+};
+
+/**
+ * hash password with sha512.
+ * @function
+ * @param {string} password - List of required fields.
+ * @param {string} salt - Data to be validated.
+ */
+var sha512 = function(password, salt){
+    var hash = crypto.createHmac('sha512', salt); /** Hashing algorithm sha512 */
+    hash.update(password);
+    var value = hash.digest('hex');
+    return {
+        salt:salt,
+        passwordHash:value
+    };
+};
+
+function saltHashPassword(userpassword) {
+    var salt = genRandomString(16); /** Gives us salt of length 16 */
+    var passwordData = sha512(userpassword, salt);
+    return ({
+        salt: salt,
+        passHash: passwordData.passwordHash
+    })
+}
+
+var verifyPass = function(password, salt, checkPass){
+    var comparePass = sha512(password, salt).passwordHash;
+    return comparePass === checkPass;
+}
+
+var verifyUser = function(username){
```

Comment:
Do we need this for anything? Username validation on register?

Reply:
Good point, I just threw it in as a just-in-case. Maybe if we want to prevent weird symbolic usernames, but that's not a big deal + we already have duplication protection